resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsSnapshot.scala

```diff
@@ -61,7 +61,22 @@ object ExecutorPodsSnapshot extends Logging {
 
   private def toStatesByExecutorId(executorPods: Seq[Pod]): Map[Long, ExecutorPodState] = {
     executorPods.map { pod =>
-      (pod.getMetadata.getLabels.get(SPARK_EXECUTOR_ID_LABEL).toLong, toState(pod))
+      pod.getMetadata.getLabels.get(SPARK_EXECUTOR_ID_LABEL) match {
+        case "EXECID" | null =>
+          // The pod has been created by something other than Spark and we should process
+          // pod the name to get the ID instead of the label.
+          val execIdRE = ".+-([0-9]+)".r
```

Comment:
Given the assumption, this RE looks fragile to me. What happens if the assumption is broken?
> The pod has been created by something other than Spark

Reply:
So if a pod is created without an exec label and the podname doesn't let us infer the exec ID we'll just throw an exception since there is not a way I could think of to recover gracefully.

Reply:
I'm worrying about picking up a non-execID number. What happens there are two numbers fields in those strings?

Reply:
So the tests actually do that currently. Since the prefix (`.+`) is "greedy" ( https://docs.oracle.com/javase/tutorial/essential/regex/quant.html ) matching it'll only match the last numbers after the -.

Reply:
Given our conversation around dynamically applying the label I'll do that instead and we can drop the re :)